src/app/my-info/(account)/reviews/_components/MyReviewProduct.tsx

```diff
@@ -1,52 +1,131 @@
 'use client';
 
+import { deleteUserProductReview } from '@/api/productReviewAPI';
+import { Dialog, Modal } from '@/components';
+import WriteEditModal from '@/components/WriteEditModal/WriteEditModal';
+
 import { formatDateToString } from '@/libs/formatDateToString';
-import { ProductType } from '@/types/ProductTypes';
+
+import type { ReviewDto } from '@/types/ProductReviewTypes';
+
+import { QueryClient, useMutation } from '@tanstack/react-query';
 import classNames from 'classnames/bind';
 import Image from 'next/image';
 import Link from 'next/link';
+import { useState } from 'react';
+import { toast } from 'react-toastify';
+
+import { ProductType } from '@/types/ProductTypes';
 import styles from './MyReviewProduct.module.scss';
 
 const cn = classNames.bind(styles);
 
 interface MyReviewProductProps {
-  productId: number;
-  updatedAt: Date;
-  switchOption: string;
+  reviewData: ReviewDto;
   productData?: ProductType;
 }
 
-export default function MyReviewProduct({ productId, updatedAt, switchOption, productData }: MyReviewProductProps) {
-  if (!productData) {
-    return <div className={cn('no-product-data')}>상품 정보를 불러올 수 없습니다.</div>;
-  }
+const defaultProductData = {
+  name: '상품 정보 없음',
+  thubmnailList: [{ id: 0, imgUrl: '/public/images/kedeukProfile.png' }],
+  categoryName: 'keyboard',
+};
+
+export default function MyReviewProduct({ reviewData, productData }: MyReviewProductProps) {
+  const { name: productName, thubmnailList, categoryName } = productData || defaultProductData;
+  const { orderId, productId, id: reviewId, updatedAt, switchOption } = reviewData;
+  const productImgUrl = thubmnailList[0].imgUrl;
+
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
 
-  const { name, thubmnailList, categoryName } = productData;
+  const queryClient = new QueryClient();
```

Comment:
```suggestion
 const queryClient = useQueryClient();
```
잘 알지는 못하지만 기존에 쓰던거 받아오려면 이렇게 써야 성능적으로 이점이 있는걸로 알고 있습니다! (보경님이랑 도원님이 암말도 안하셨네용,,사실 잘모름,,)